### PR TITLE
withStoryContainer: removes addon-knobs since we've migrated to controls

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR removes addon-knobs since we've migrated all our stories to use controls.
**Which issue(s) this PR fixes**:

Relates to #34417

### CODE REVIEW

File: packages/grafana-ui/src/utils/storybook/withStoryContainer.tsx [hunk 1]
```diff
@@ -44,38 +43,8 @@ const StoryContainer: React.FC<{ width?: number; height?: number; showBoundaries
...
-    },
-    CONTAINER_GROUP
-  );
   return (
-    <StoryContainer
-      width={fullWidthContainer ? undefined : containerWidth}
-      height={containerHeight}
-      showBoundaries={containerBoundary}
-    >
+    <StoryContainer width={300} height={0} showBoundaries={false}>
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/utils/storybook/withStoryContainer.tsx:47]: I don't know how the clean up plan for Storybook looks like so this question might be out of place, sorry if that is the case. With this change you're removing the possibility for the user of Storybook to resize the story. Not sure this is an important feature for us but we should check if this is needed. 

Is there a way to use Storybook controls to achieve the same resize feature? I can see that the decorators now have a second argument passed to them which is the StoryContext. Can this be used somehow?

```ts
export const withStoryContainer = (story: RenderFunction, context: StoryContext) => {
  return (
    <StoryContainer width={300} height={0} showBoundaries={false}>
      {story()}
    </StoryContainer>
  );
};
```

image

**Review by hugohaggmark — COMMENTED** [review]
Looks good, I left a comment that you might look into.

File: packages/grafana-ui/src/utils/storybook/withStoryContainer.tsx (hunk 1 above)
Comment (thisisobate) [reply packages/grafana-ui/src/utils/storybook/withStoryContainer.tsx:47]: Great catch! Would investigate and see how we could use the StoryContext here.

**Comment by thisisobate** [comment]
Since we use the `withStoryContainer` in the `FieldArray` story, I refactored the story to have the ability to resize the story.
You can take a look: https://github.com/grafana/grafana/pull/35820

**Comment by thisisobate** [comment]
For the `Form` story, I think we shouldn't bother adding controls to it since it doesn't support controls by default.

**Comment by hugohaggmark** [comment]
@thisisobate I think you should close those PRs and include those fixes in this PR. Makes it a lot easier to test.

**Comment by thisisobate** [comment]
Since I've gotten reviews on other PRs, I will just tackle them right away and merge them to this PR. Hope that's ok with you @hugohaggmark

File: packages/grafana-ui/src/components/Layout/Layout.story.tsx [hunk 2]
```diff
@@ -28,8 +25,13 @@ export default {
     align: 'center',
     wrap: false,
     width: '100%',
+    ContainerWidth: 300,
+    ContainerHeight: 0,
+    showBoundaries: false,
   },
   argTypes: {
+    ContainerWidth: { control: { type: 'range', min: 100, max: 500, step: 10 } },
+    ContainerHeight: { control: { type: 'range', min: 100, max: 500, step: 10 } },
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Layout/Layout.story.tsx:34]: should these have uppercase naming?

File: packages/grafana-ui/src/utils/storybook/withStoryContainer.tsx [hunk 3]
```diff
@@ -43,38 +43,12 @@ const StoryContainer: React.FC<{ width?: number; height?: number; showBoundaries
...
-    },
-    CONTAINER_GROUP
-  );
+export const withStoryContainer = (story: RenderFunction, context: StoryContext) => {
   return (
     <StoryContainer
-      width={fullWidthContainer ? undefined : containerWidth}
-      height={containerHeight}
-      showBoundaries={containerBoundary}
+      width={context.args.containerWidth}
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/utils/storybook/withStoryContainer.tsx:49]: width prop isn't working because you're using lower case `containerWidth` here

**Review by hugohaggmark — CHANGES_REQUESTED** [review]
Looks better, but the width isn't working because of naming.

File: packages/grafana-ui/src/components/Layout/Layout.story.tsx (hunk 2 above)
Comment (thisisobate) [reply packages/grafana-ui/src/components/Layout/Layout.story.tsx:34]: would fix it right away

File: packages/grafana-ui/src/utils/storybook/withStoryContainer.tsx (hunk 3 above)
Comment (thisisobate) [reply packages/grafana-ui/src/utils/storybook/withStoryContainer.tsx:49]: Great catch! pushing a fix right away

**Review by hugohaggmark — APPROVED** [review]
Looks good 🎉

**Pull request merged**